### ISSUE TITLE
Sequential vs Functional API. Drop in model accuracy

### ISSUE THREAD

**Issue by ujohn33** [philipperemy/cond_rnn#23]
Hi, I am doing research on time-series forecasting for electric load.  I am quite new to tensorflow, so please excuse me if make errors.
I came across your library and thought it was very interesting! I am building a 24-ahead MIMO forecasting model. I have noticed that pure LSTM may disregard features that I considered important, so I have experimented with this framework.
I have tried using conditional RNN cell on the meter data using the examples in the repository. For auxillary feautures I used features of the predicted time step (e.g. one hot encoded day of week) and important lags as features. 
First, writing and fitting the sequential model:

```
ConditionalRNN = Sequential(layers=[ConditionalRNN(32, cell='LSTM'),
                                    Dense(HORIZON)])
ConditionalRNN.compile(optimizer=tf.optimizers.Adam(), loss='mse', metrics=[tf.metrics.MeanSquaredError()])
history = ConditionalRNN.fit([train_inputs['X'],train_c1,train_c2], train_inputs['target'], batch_size=32, epochs=MAX_EPOCHS, validation_split=0.15, callbacks=[early_stopping], verbose=1)
```

The model works fine and shows better accuracy than the LSTM with the full data passed through it. 
However, I was getting this warning from tensoflow:

_WARNING:tensorflow:Layers in a Sequential model should only have a single input tensor, but we receive a <class 'tuple'> input: (<tf.Tensor 'IteratorGetNext:0' shape=(None, 24, 5) dtype=float32>, <tf.Tensor 'IteratorGetNext:1' shape=(None, 8) dtype=float64>, <tf.Tensor 'IteratorGetNext:2' shape=(None, 2) dtype=float64>)
Consider rewriting this model with the Functional API._

I have rewritten the same model in a functional API way:
```
i = Input(shape=[HORIZON, 5], name='time-series')
c1 = Input(shape=[8], name='one-hot')
c2 = Input(shape=[2], name='lags-as-features')
x = ConditionalRNN(32, cell='LSTM', name='cond_rnn_0')([i, c1, c2])
#x = Dense(HORIZON)(x)
#ConditionalRNN = Model(inputs=[i, c1, c2], outputs=[x])
```
This way there has been no warnings but the accuracy has significantly dropped down. 
Do you have any explanation on why it might be the case?

**Comment by adaj** [comment]
I am also receiving this warning but ignoring it. One thing that I had in mind to test is to specify `input_shape`, but I am not sure how to specify that given the input structure of `ConditionalRNN`.

**Comment by philipperemy** [comment]
Glad to see that CondRNN could give you a boost in accuracy!

First, regarding the message, I think this is fine. Conceptually, a Sequential model is not expected to receive multiple inputs. Even though Keras is complaining, it's still doing the job correctly.

```
WARNING:tensorflow:Layers in a Sequential model should only have a single input tensor, but we receive a <class 'tuple'> input: (<tf.Tensor 'IteratorGetNext:0' shape=(None, 24, 5) dtype=float32>, <tf.Tensor 'IteratorGetNext:1' shape=(None, 8) dtype=float64>, <tf.Tensor 'IteratorGetNext:2' shape=(None, 2) dtype=float64>)
Consider rewriting this model with the Functional API.
```

I don't see any errors in your code. I checked, and both Sequential are Functional output the same code.

On the `dummy_stations_example.py` example, I added the Functional API code and I ran two simple training:

*Sequential*
```
788/788 - 14s - loss: 0.0445 - val_loss: 0.0171
Epoch 2/10
788/788 - 13s - loss: 0.0160 - val_loss: 0.0156
Epoch 3/10
788/788 - 14s - loss: 0.0155 - val_loss: 0.0154
Epoch 4/10
788/788 - 16s - loss: 0.0151 - val_loss: 0.0155
Epoch 5/10
788/788 - 13s - loss: 0.0150 - val_loss: 0.0150
Epoch 6/10
788/788 - 13s - loss: 0.0150 - val_loss: 0.0148
Epoch 7/10
788/788 - 13s - loss: 0.0149 - val_loss: 0.0147
Epoch 8/10
788/788 - 14s - loss: 0.0150 - val_loss: 0.0149
Epoch 9/10
788/788 - 14s - loss: 0.0149 - val_loss: 0.0152
Epoch 10/10
788/788 - 14s - loss: 0.0149 - val_loss: 0.0146
```

*Functional*
```
Functional API
Epoch 1/10
788/788 - 14s - loss: 0.0317 - val_loss: 0.0165
Epoch 2/10
788/788 - 14s - loss: 0.0160 - val_loss: 0.0156
Epoch 3/10
788/788 - 14s - loss: 0.0153 - val_loss: 0.0149
Epoch 4/10
788/788 - 15s - loss: 0.0150 - val_loss: 0.0153
Epoch 5/10
788/788 - 14s - loss: 0.0150 - val_loss: 0.0155
Epoch 6/10
788/788 - 13s - loss: 0.0149 - val_loss: 0.0151
Epoch 7/10
788/788 - 14s - loss: 0.0148 - val_loss: 0.0150
Epoch 8/10
788/788 - 13s - loss: 0.0149 - val_loss: 0.0147
Epoch 9/10
788/788 - 13s - loss: 0.0148 - val_loss: 0.0147
Epoch 10/10
788/788 - 16s - loss: 0.0148 - val_loss: 0.0146
```

They look pretty similar in terms of performance/loss.

**Comment by philipperemy** [comment]
@adaj yes from my experience, you can safely ignore this warning. Keras still does the job well.
